ElastosDIDSDK/DIDDocument.swift

```diff
@@ -687,34 +722,69 @@ public class DIDDocument {
         return DIDDocumentBuilder(self)
     }
 
-    public func sign(using storePassword: String, for data: Data...) throws -> String {
-        return try sign(self.defaultPublicKey, storePassword, data)
+    public func sign(withId: DIDURL, using storepass: String, for data: Data...) throws -> String {
+        try sign(withId, storepass, data)
     }
 
-    public func sign(withId: DIDURL, using storePassword: String, for data: Data...) throws -> String {
-        return try sign(withId, storePassword, data)
+    public func sign(withId: String, using storepass: String, for data: Data...) throws -> String {
+        return try sign(try DIDURL(self.subject, withId), storepass, data)
     }
 
-    public func sign(withId: String, using storePassword: String, for data: Data...) throws -> String {
-        return try sign(try DIDURL(self.subject, withId), storePassword, data)
+    public func sign(using storepass: String, for data: Data...) throws -> String {
+        return try sign(self.defaultPublicKey, storepass, data)
     }
 
-    func sign(_ id: DIDURL, _ storePassword: String, _ data: [Data]) throws -> String {
+    func sign(_ id: DIDURL, _ storepass: String, _ data: [Data]) throws -> String {
         guard data.count > 0 else {
             throw DIDError.illegalArgument()
         }
-        guard !storePassword.isEmpty else {
+        guard !storepass.isEmpty else {
+            throw DIDError.illegalArgument()
+        }
+
+        var cinputs: [CVarArg] = []
+        var capacity: Int = 0
+        data.forEach { data in
+            let json = String(data: data, encoding: .utf8)
+            if json != "" {
+                let cjson = json!.toUnsafePointerInt8()!
+                cinputs.append(cjson)
+                cinputs.append(json!.count)
+                capacity += json!.count * 3
+            }
+        }
+
+        let c_inputs = getVaList(cinputs)
+        let count = cinputs.count / 2
+        _capacity = capacity
+        // digest
+        let cdigest = UnsafeMutablePointer<UInt8>.allocate(capacity: capacity)
+        let size = sha256v_digest(cdigest, Int32(count), c_inputs)
+        let cdigestPointerToArry: UnsafeBufferPointer<UInt8> = UnsafeBufferPointer(start: cdigest, count: size)
+        let digest = Data(buffer: cdigestPointerToArry)
+
+        return try signDigest(id, storepass, digest)
+    }
+
+    public func signDigest(_ id: DIDURL, _ storepass: String, _ digest: Data) throws -> String {
```

Comment:
I guess we need APIs to support variables data parameters.
func signDigest(id, storePassword, data: Data ...)